src/jupyter_contrib_nbextensions/nbextensions/snippets/main.js

```diff
@@ -0,0 +1,72 @@
+// Code cell snippets
+
+define([
+    'jquery',
+    'base/js/namespace',
+    'base/js/dialog',
+    'base/js/utils',
+    'services/config'
+], function($, Jupyter, dialog, utils, configmod) {
+    "use strict";
+
+    // create config object to load parameters
+    var base_url = utils.get_body_data("baseUrl");
+    var config = new configmod.ConfigSection('notebook', {base_url: base_url});
+
+    config.loaded.then(function() {
+        var dropdown = $("<select></select>").attr("id", "snippet_picker")
+                                             .css("margin-left", "0.75em")
+                                             .attr("class", "form-control select-xs")
+                                             .change(insert_cell);
+        Jupyter.toolbar.element.append(dropdown);
+    });
+
+    // will be called when the nbextension is loaded
+    function load_extension() {
+        config.load(); // trigger loading config parameters
+
+        $.getJSON("/nbextensions/snippets/snippets.json", function(data) {
```

Comment:
Also, the `$.getJSON` callback relies on the config.loaded already having added the select, so it might be better to chain them directly. I should have mentioned these before merging, really, sorry!